policy/nomad/source.go

```diff
@@ -71,56 +85,70 @@ func (s *Source) MonitorIDs(ctx context.Context, req policy.MonitorIDsReq) {
 	q := &api.QueryOptions{WaitTime: 5 * time.Minute, WaitIndex: 1}
 
 	for {
+		var (
+			policies []*api.ScalingPolicyListStub
+			meta     *api.QueryMeta
+			err      error
+		)
+
+		// Perform a blocking query on the Nomad API that returns a stub list
+		// of scaling policies. The call is done in a goroutine so we can
+		// still listen for the context closing or a reload request.
+		blockingQueryCompleteCh := make(chan struct{})
+		go func() {
+			// Obtain a handler now so we can release the lock before starting
+			// the blocking query.
+			s.nomadLock.RLock()
+			scaling := s.nomad.Scaling()
+			s.nomadLock.RUnlock()
+
+			policies, meta, err = scaling.ListPolicies(q)
```

Comment:
i'm still annoyed that we can't send a context into these blocking queries...

Reply:
:nod: